apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs

```diff
@@ -292,5 +291,37 @@ private ImmutableStack<TraceSpan> TraceStack
             }
         }
 #endif
+
+        // TODO: Remove when https://github.com/dotnet/roslyn/issues/12255 is fixed and go back to using System.Collections.Immutable
```

Comment:
Not sure even then, to be honest. If we can avoid an extra dependency just by having 30 lines of code, that could easily simplify things in the future.